apps/prairielearn/src/middlewares/AuthzAccessMismatch.tsx

```diff
@@ -1,14 +1,32 @@
 import _ from 'lodash';
-import OverlayTrigger from 'react-bootstrap/OverlayTrigger';
-import Tooltip from 'react-bootstrap/Tooltip';
 
 import { removeCookieClient, setCookieClient } from '../lib/client/cookie.js';
 import type { PageContext } from '../lib/client/page-context.js';
 import type { StaffUser } from '../lib/client/safe-db-types.js';
 
+// These keys can be used as part of permission checks.
+export type CheckablePermissionKeys = Extract<
+  | 'is_administrator'
+  | 'has_course_permission_preview'
+  | 'has_course_permission_view'
+  | 'has_course_permission_edit'
+  | 'has_course_permission_own'
+  | 'has_course_instance_permission_view'
+  | 'has_course_instance_permission_edit'
+  | 'has_student_access'
+  | 'has_student_access_with_enrollment',
+  keyof PageContext['authz_data']
+>;
```

Comment:
This is updated to be more precise than just "any key in `authz_data`. The previous implementation would have allowed keys like `user` that wouldn't make sense to check.